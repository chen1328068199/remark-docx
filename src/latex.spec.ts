import { it, describe, expect } from "vitest";
import { parseLatex } from "./latex";

const parse = (str: string) => parseLatex(str);

describe("parse", () => {
  [
    "L = \\frac{1}{2} \\rho v^2 S C_L",
    "i hbar \\frac{partial psi}{partial t} = H psi(x,t)",
    "\\f\\relax{x} = \\int_{-\\infty}^\\infty\\f\\hat\\xi\\,e^{2 \\pi i \\xi x}\\,d\\xi",
    "\\frac{a+b}{c+d}",
    "\\sqrt[30]{a+\\sqrt{b+c}}",
    "\\sum_{k=1}^{n} k=\frac{n(n+1)}{2}",
  ].forEach((s) => {
    it(s, () => {
      expect(parse(s)).toMatchSnapshot();
    });
  });

  describe("command", () => {
    it("greek1", () => {
      expect(
        parse(
          "\\alpha\\beta\\gamma\\delta\\epsilon\\zeta\\eta\\theta\\iota\\kappa\\lambda\\mu\\nu\\xi o\\pi\\rho\\sigma\\tau\\upsilon\\phi\\chi\\psi\\omega",
        ),
      ).toMatchSnapshot();
    });
    it("greek2", () => {
      expect(
        parse("\\varepsilon\\vartheta\\varrho\\varsigma\\varphi"),
      ).toMatchSnapshot();
    });
    it("greek3", () => {
      expect(
        parse(
          "\\Gamma\\Delta\\Theta\\Lambda\\Xi\\Pi\\Sigma\\Upsilon\\Phi\\Psi\\Omega",
        ),
      ).toMatchSnapshot();
    });
  });
});
