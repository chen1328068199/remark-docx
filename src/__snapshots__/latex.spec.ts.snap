// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parse > \\f\\relax{x} = \\int_{-\\infty}^\\infty\\f\\hat\\xi\\,e^{2 \\pi i \\xi x}\\,d\\xi 1`] = `
[
  [
    MathRun {
      "root": [
        MathText {
          "root": [
            "f",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "relax",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "x",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "=",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathSuperScript {
      "root": [
        MathSuperScriptProperties {
          "root": [],
          "rootKey": "m:sSupPr",
        },
        MathBase {
          "root": [
            MathSubScript {
              "root": [
                MathSubScriptProperties {
                  "root": [],
                  "rootKey": "m:sSubPr",
                },
                MathBase {
                  "root": [
                    MathRun {
                      "root": [
                        MathText {
                          "root": [
                            "∫",
                          ],
                          "rootKey": "m:t",
                        },
                      ],
                      "rootKey": "m:r",
                    },
                  ],
                  "rootKey": "m:e",
                },
                MathSubScriptElement {
                  "root": [
                    MathRun {
                      "root": [
                        MathText {
                          "root": [
                            "-",
                          ],
                          "rootKey": "m:t",
                        },
                      ],
                      "rootKey": "m:r",
                    },
                    MathRun {
                      "root": [
                        MathText {
                          "root": [
                            "∞",
                          ],
                          "rootKey": "m:t",
                        },
                      ],
                      "rootKey": "m:r",
                    },
                  ],
                  "rootKey": "m:sub",
                },
              ],
              "rootKey": "m:sSub",
            },
          ],
          "rootKey": "m:e",
        },
        MathSuperScriptElement {
          "root": [
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "∞",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
          ],
          "rootKey": "m:sup",
        },
      ],
      "rootKey": "m:sSup",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "f",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "hat",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "ξ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            ",",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathSuperScript {
      "root": [
        MathSuperScriptProperties {
          "root": [],
          "rootKey": "m:sSupPr",
        },
        MathBase {
          "root": [
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "e",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
          ],
          "rootKey": "m:e",
        },
        MathSuperScriptElement {
          "root": [
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "2",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "π",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "i",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "ξ",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "x",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
          ],
          "rootKey": "m:sup",
        },
      ],
      "rootKey": "m:sSup",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            ",",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "d",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "ξ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
  ],
]
`;

exports[`parse > \\frac{a+b}{c+d} 1`] = `
[
  [
    MathFraction {
      "root": [
        MathNumerator {
          "root": [
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "a",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "+",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "b",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
          ],
          "rootKey": "m:num",
        },
        MathDenominator {
          "root": [
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "c",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "+",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "d",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
          ],
          "rootKey": "m:den",
        },
      ],
      "rootKey": "m:f",
    },
  ],
]
`;

exports[`parse > \\sqrt[30]{a+\\sqrt{b+c}} 1`] = `
[
  [
    MathRadical {
      "root": [
        MathRadicalProperties {
          "root": [],
          "rootKey": "m:radPr",
        },
        MathDegree {
          "root": [
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "3",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "0",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
          ],
          "rootKey": "m:deg",
        },
        MathBase {
          "root": [
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "a",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "+",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "sqrt",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
          ],
          "rootKey": "m:e",
        },
      ],
      "rootKey": "m:rad",
    },
  ],
]
`;

exports[`parse > \\sum_{k=1}^{n} k=rac{n(n+1)}{2} 1`] = `
[
  [
    MathSuperScript {
      "root": [
        MathSuperScriptProperties {
          "root": [],
          "rootKey": "m:sSupPr",
        },
        MathBase {
          "root": [
            MathSubScript {
              "root": [
                MathSubScriptProperties {
                  "root": [],
                  "rootKey": "m:sSubPr",
                },
                MathBase {
                  "root": [
                    MathSum {
                      "root": [
                        MathNAryProperties {
                          "root": [
                            MathAccentCharacter {
                              "root": [
                                MathAccentCharacterAttributes {
                                  "root": {
                                    "accent": "∑",
                                  },
                                  "rootKey": "_attr",
                                  "xmlKeys": {
                                    "accent": "m:val",
                                  },
                                },
                              ],
                              "rootKey": "m:chr",
                            },
                            MathLimitLocation {
                              "root": [
                                MathLimitLocationAttributes {
                                  "root": {
                                    "value": "undOvr",
                                  },
                                  "rootKey": "_attr",
                                  "xmlKeys": {
                                    "value": "m:val",
                                  },
                                },
                              ],
                              "rootKey": "m:limLoc",
                            },
                            MathSuperScriptHide {
                              "root": [
                                MathSuperScriptHideAttributes {
                                  "root": {
                                    "hide": 1,
                                  },
                                  "rootKey": "_attr",
                                  "xmlKeys": {
                                    "hide": "m:val",
                                  },
                                },
                              ],
                              "rootKey": "m:supHide",
                            },
                            MathSubScriptHide {
                              "root": [
                                MathSubScriptHideAttributes {
                                  "root": {
                                    "hide": 1,
                                  },
                                  "rootKey": "_attr",
                                  "xmlKeys": {
                                    "hide": "m:val",
                                  },
                                },
                              ],
                              "rootKey": "m:subHide",
                            },
                          ],
                          "rootKey": "m:naryPr",
                        },
                        MathBase {
                          "root": [],
                          "rootKey": "m:e",
                        },
                      ],
                      "rootKey": "m:nary",
                    },
                  ],
                  "rootKey": "m:e",
                },
                MathSubScriptElement {
                  "root": [
                    MathRun {
                      "root": [
                        MathText {
                          "root": [
                            "k",
                          ],
                          "rootKey": "m:t",
                        },
                      ],
                      "rootKey": "m:r",
                    },
                    MathRun {
                      "root": [
                        MathText {
                          "root": [
                            "=",
                          ],
                          "rootKey": "m:t",
                        },
                      ],
                      "rootKey": "m:r",
                    },
                    MathRun {
                      "root": [
                        MathText {
                          "root": [
                            "1",
                          ],
                          "rootKey": "m:t",
                        },
                      ],
                      "rootKey": "m:r",
                    },
                  ],
                  "rootKey": "m:sub",
                },
              ],
              "rootKey": "m:sSub",
            },
          ],
          "rootKey": "m:e",
        },
        MathSuperScriptElement {
          "root": [
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "n",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
          ],
          "rootKey": "m:sup",
        },
      ],
      "rootKey": "m:sSup",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "k",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "=",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "r",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "a",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "c",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "n",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "(",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "n",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "+",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "1",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            ")",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "2",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
  ],
]
`;

exports[`parse > L = \\frac{1}{2} \\rho v^2 S C_L 1`] = `
[
  [
    MathRun {
      "root": [
        MathText {
          "root": [
            "L",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "=",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathFraction {
      "root": [
        MathNumerator {
          "root": [
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "1",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
          ],
          "rootKey": "m:num",
        },
        MathDenominator {
          "root": [
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "2",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
          ],
          "rootKey": "m:den",
        },
      ],
      "rootKey": "m:f",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "ρ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathSuperScript {
      "root": [
        MathSuperScriptProperties {
          "root": [],
          "rootKey": "m:sSupPr",
        },
        MathBase {
          "root": [
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "v",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
          ],
          "rootKey": "m:e",
        },
        MathSuperScriptElement {
          "root": [
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "2",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
          ],
          "rootKey": "m:sup",
        },
      ],
      "rootKey": "m:sSup",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "S",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathSubScript {
      "root": [
        MathSubScriptProperties {
          "root": [],
          "rootKey": "m:sSubPr",
        },
        MathBase {
          "root": [
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "C",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
          ],
          "rootKey": "m:e",
        },
        MathSubScriptElement {
          "root": [
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "L",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
          ],
          "rootKey": "m:sub",
        },
      ],
      "rootKey": "m:sSub",
    },
  ],
]
`;

exports[`parse > command > greek1 1`] = `
[
  [
    MathRun {
      "root": [
        MathText {
          "root": [
            "α",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "β",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "γ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "δ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "ϵ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "ζ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "η",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "θ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "ι",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "κ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "λ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "μ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "ν",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "ξ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "o",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "π",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "ρ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "σ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "τ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "υ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "ϕ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "χ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "ψ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "ω",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
  ],
]
`;

exports[`parse > command > greek2 1`] = `
[
  [
    MathRun {
      "root": [
        MathText {
          "root": [
            "ε",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "ϑ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "ϱ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "ς",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "φ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
  ],
]
`;

exports[`parse > command > greek3 1`] = `
[
  [
    MathRun {
      "root": [
        MathText {
          "root": [
            "Γ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "Δ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "Θ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "Λ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "Ξ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "Π",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "Σ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "Υ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "Φ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "Ψ",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "Ω",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
  ],
]
`;

exports[`parse > i hbar \\frac{partial psi}{partial t} = H psi(x,t) 1`] = `
[
  [
    MathRun {
      "root": [
        MathText {
          "root": [
            "i",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "h",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "b",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "a",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "r",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathFraction {
      "root": [
        MathNumerator {
          "root": [
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "p",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "a",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "r",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "t",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "i",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "a",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "l",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "p",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "s",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "i",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
          ],
          "rootKey": "m:num",
        },
        MathDenominator {
          "root": [
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "p",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "a",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "r",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "t",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "i",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "a",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "l",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
            MathRun {
              "root": [
                MathText {
                  "root": [
                    "t",
                  ],
                  "rootKey": "m:t",
                },
              ],
              "rootKey": "m:r",
            },
          ],
          "rootKey": "m:den",
        },
      ],
      "rootKey": "m:f",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "=",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "H",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "p",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "s",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "i",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "(",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "x",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            ",",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            "t",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
    MathRun {
      "root": [
        MathText {
          "root": [
            ")",
          ],
          "rootKey": "m:t",
        },
      ],
      "rootKey": "m:r",
    },
  ],
]
`;
